lib/handlers/pools/pool-caching/v2/v2-debug-caching-pool-provider.ts

```diff
@@ -0,0 +1,26 @@
+import { IV2PoolProvider, V2PoolAccessor } from '@uniswap/smart-order-router'
+import { Token } from '@uniswap/sdk-core'
+import { ProviderConfig } from '@uniswap/smart-order-router/build/main/providers/provider'
+
+export class V2DebugCachingPoolProvider implements IV2PoolProvider {
+  cachingPoolProvider: IV2PoolProvider
+  poolProvider: IV2PoolProvider
+
+  constructor(cachingPoolProvider: IV2PoolProvider, poolProvider: IV2PoolProvider) {
+    this.cachingPoolProvider = cachingPoolProvider
+    this.poolProvider = poolProvider
+  }
+
+  getPoolAddress(tokenA: Token, tokenB: Token): { poolAddress: string; token0: Token; token1: Token } {
+    // this is in-memory operation, can just use the existing non-caching pool provider
+    return this.poolProvider.getPoolAddress(tokenA, tokenB)
+  }
+
+  getPools(tokenPairs: [Token, Token][], providerConfig?: ProviderConfig): Promise<V2PoolAccessor> {
+    if (providerConfig?.debugRouting) {
+      return this.cachingPoolProvider.getPools(tokenPairs, providerConfig)
```

Comment:
How do we plan on testing this?

Reply:
I plan to test this via debug query string param:

curl --request GET '<URL>?tokenInAddress=0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2&tokenInChainId=1&tokenOutAddress=0x1f9840a85d5af5bf1d1762f925bdaddc4201f984&tokenOutChainId=1&amount=100&type=exactIn&unicornSecret=0x0&debugRoutingConfig={"v2PoolSelection.topN":0,"v2PoolSelection.topNDirectSwaps":2,"v2PoolSelection.topNTokenInOut":2,"v2PoolSelection.topNSecondHop":0,"v2PoolSelection.topNWithEachBaseToken":0,"v2PoolSelection.topNWithBaseToken":0,"v3PoolSelection.topN":0,"v3PoolSelection.topNDirectSwaps":2,"v3PoolSelection.topNTokenInOut":2,"v3PoolSelection.topNSecondHop":0,"v3PoolSelection.topNWithEachBaseToken":0,"v3PoolSelection.topNWithBaseToken":0,"maxSwapsPerPath":2,"maxSplits":2,"distributionPercent":20,"writeToCachedRoutes":false,"debugRouting":true}'

and from my perf testing script. My perf testing script already account for the latencies difference, so it's easy to compare apple-to-apple if this gets shipped into prod, and we can collect to perf data from prod traffic volume.